File: java/program/Memory.java
class Memory extends Hardware {
    private int size, speed;
    public Memory(String name, double price, int warrantyPeriod, String brand, int size, int speed) {
        super(name, price, warrantyPeriod, brand);
        this.size = size;
        this.speed = speed;
    }
    public int getSize() { return size; }
    public int getSpeed() { return speed; }

    // setter
    public void setSize(int size) { this.size = size; }
    public void setSpeed(int speed) { this.speed = speed; }
    
}

File: cpp/program/Memory.cpp
#pragma once
#include <iostream>
#include <vector>
using namespace std;
#include "Hardware.cpp"

class Memory : public Hardware {
    private:
        int size;  // Kapasitas (GB)
        int speed; // Kecepatan (MHz)
    
    public:
        Memory(string n, float p, int w, string b, int sz, int spd)
            : Hardware(n, p, w, b), size(sz), speed(spd) {}
    
        // Getters
        int getSize() const { return size; }
        int getSpeed() const { return speed; }
    
        // Setters
        void setSize(int sz) { size = sz; }
        void setSpeed(int spd) { speed = spd; }
    };